### PR TITLE
Async document v1

### PR DESCRIPTION
**Pull request by jonmv**
@bjorncs please review.

### CODE REVIEW

**Comment by jonmv** [comment]
I have faith in this last commit.

File: vespaclient-container-plugin/src/test/java/com/yahoo/document/restapi/resource/DocumentV1ApiTest.java [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+                           "}",
+                           response.readAll());
+            assertEquals(200, response.getStatus());
+            assertEquals(new DocumentRemove(doc2.getId()), executor.lastOperation());
+            assertEquals(parameters().withRoute("route"), executor.lastParameters());
+        }
+    }
+
+    void assertSameJson(String expected, String actual) {
+        ByteArrayOutputStream expectedPretty = new ByteArrayOutputStream();
```
Comment (bjorncs) [inline vespaclient-container-plugin/src/test/java/com/yahoo/document/restapi/resource/DocumentV1ApiTest.java:360]: Use the JSON assert helper in `testutils`

File: documentapi/src/main/java/com/yahoo/documentapi/local/LocalDocumentAccess.java [hunk 2]
```diff
@@ -94,7 +95,7 @@ public SubscriptionSession openSubscription(SubscriptionParameters parameters) {
      *   session.put(documentPut);
      *   session.get(documentId);
      *                                     // Operations wait for this thread to arrive at "phaser"
-     *   phaser.arrive();                  // Let operations send their responses
+     *   phaser.arriveAndAwaitAdvance();   // Let operations send their responses
```
Comment (bjorncs) [inline documentapi/src/main/java/com/yahoo/documentapi/local/LocalDocumentAccess.java:98]: Unintended change?

**Review by bjorncs — APPROVED** [review]

File: vespaclient-container-plugin/src/test/java/com/yahoo/document/restapi/resource/DocumentV1ApiTest.java (hunk 1 above)
Comment (jonmv) [reply vespaclient-container-plugin/src/test/java/com/yahoo/document/restapi/resource/DocumentV1ApiTest.java:360]: Nice, thanks.

File: documentapi/src/main/java/com/yahoo/documentapi/local/LocalDocumentAccess.java (hunk 2 above)
Comment (jonmv) [reply documentapi/src/main/java/com/yahoo/documentapi/local/LocalDocumentAccess.java:98]: No, it was wrong.

**Pull request merged**